pkg/api/microservices/microservices_test.go

```diff
@@ -0,0 +1,159 @@
+	}
+	// GIVEN empty model binding pairs, managed clusters and fake kubernetes clients.
+	var clients kubernetes.Interface = k8sfake.NewSimpleClientset()
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+
+	type args struct {
+		w http.ResponseWriter
+		r *http.Request
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{
+			name:    "verifyReturnAllMicroservices",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var myMicroservices []Microservice
+			Init(tt.listers)
+			resp := httptest.NewRecorder()
+			ReturnAllMicroservices(resp, httptest.NewRequest("GET", "/microservices", nil))
+			assert.Equal(t, http.StatusOK, resp.Code, "expect the http return code to be http.StatusOk")
+			json.NewDecoder(resp.Body).Decode(&myMicroservices)
+			assert.Len(t, myMicroservices, 2, "expect returned MicroServices Array to have 2 entries")
+			assert.Equal(t, "HelidonApp1", myMicroservices[0].Name, "expect Helidon App Name to be HelidonApp1")
+			assert.Equal(t, "cluster1", myMicroservices[0].Cluster, "expect Helidon App Cluster Name to be cluster1")
+			assert.Equal(t, "defaultOnMC1", myMicroservices[0].Namespace, "expect Helidon App Namespace to be defaultOnMC1")
+			assert.Equal(t, "HelidonApp2", myMicroservices[1].Name, "expect Helidon App Name to be HelidonApp2")
+			assert.Equal(t, "cluster2", myMicroservices[1].Cluster, "expect Helidon App Cluster Name to be cluster2")
+			assert.Equal(t, "defaultOnMC2", myMicroservices[1].Namespace, "expect Helidon App Namespace to be defaultOnMC2")
+		})
+	}
+}
+
+// TestReturnSingleMicroservice tests that the Microservice with ID=HelidonApp1 is returned
+// GIVEN that Microservice with ID=HelidonApp1 exists
+// WHEN I call ReturnSingleMicroservice
+// THEN the Microservice with ID=HelidonApp1 should be returned
+// AND that the Microservice returned should have the correct ID and have valid values
+func TestReturnSingleMicroservice(t *testing.T) {
+	var modelBindingPairs = map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	// GIVEN empty model binding pairs, managed clusters and fake kubernetes clients.
+	var clients kubernetes.Interface = k8sfake.NewSimpleClientset()
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+
+	type args struct {
+		w http.ResponseWriter
+		r *http.Request
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{
+			name:    "verifyReturnSingleMicroservice",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			vars := map[string]string{
+				"id": "HelidonApp1",
+			}
+			var myMicroservice Microservice
+			Init(tt.listers)
+			resp := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/microservices", nil)
```

Comment:
this request should be for /microservices/HelidonApp1 or something like that, right? ReturnSingleMicroservice looks for an id variable in the pathInfo of the request.

Reply:
in fact, I'm not sure how this test is passing if that's not the case - could be a bug in the code, or I'm missing something

Reply:
see  my comments on PR 152